platform/ios/docs/guides/For Style Authors.md

```diff
@@ -190,6 +190,7 @@ In style JSON | In the SDK
 `circle` | `MGLCircleStyleLayer`
 `fill` | `MGLFillStyleLayer`
 `fill-extrusion` | `MGLFillExtrusionStyleLayer`
+`hillshade` | `MGLHillshadeStyleLayer`
```

Comment:
Unfortunately, darwin-style-code is unable to automate the entire process of adding a new layer type, so some changes are still needed before this layer type can be used on iOS and macOS:

* [ ] [Add the new files](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/ios/DEVELOPING.md#adding-a-source-code-file) to the SDKs
* [ ] [Add the new classes](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/ios/DEVELOPING.md#making-a-type-or-constant-public) to the documentation table of contents
* [ ] [Wrap the new layer type](https://github.com/mapbox/mapbox-gl-native/blob/2d15aed43c9faa875a8f625c3afc286f1175e0ce/platform/darwin/src/MGLStyle.mm#L388) in an Objective-C object in the runtime styling API
* [ ] [Refine the new class’s documentation](https://github.com/mapbox/mapbox-gl-native/blob/2d15aed43c9faa875a8f625c3afc286f1175e0ce/platform/darwin/scripts/style-spec-overrides-v8.json#L11)
* [ ] [Add an icon](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/macos/app/StyleLayerIconTransformer.m) to represent this layer type in the sidebar of macosapp
* [ ] Mention the new APIs in the changelogs

This can be tail work if necessary, since the builds are passing.

Reply:
thanks for the heads up @1ec5 – I would prefer to do this in a follow up PR because this is already a giant set of changes if thats ok

Reply:
ticketed https://github.com/mapbox/mapbox-gl-native/issues/10991